frontend/src/components/TextPopup.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  IconButton,
+  Stack,
+  SvgIcon,
+  Typography,
+} from '@mui/material';
+import ContentEditor from './ContentEditor';
+import { textSecondaryColor } from 'theme';
+import { ReactComponent as Xbtn } from 'assets/icons/x-circle.svg';
+import { ReactComponent as ArrowLogo } from 'assets/icons/chevron-down.svg';
+import { useTranslation } from 'react-i18next';
+import { StringBank } from 'strings';
+import { JSONContent } from '@tiptap/react';
+import { useState } from 'react';
+
+interface dialogProps {
+  isOpen: boolean;
+  parentSection: string;
+  newVersionName: string;
+  cancleFn: (val: boolean) => unknown;
```

Comment:
- [ ] Let's rename xFn to onX (onCancel, onComplete)
- [ ] Fix typos: cancel instead of cancle
- [ ] Remove usage of any, use either known type or unknown (for example `(value: string) => void` for completeFn)
- [ ] Do we use any variables other than sectionId? if not, we could probably just use `sectionId` prop instead of `variables`.

Reply:
the idea of "variables" is to allow the textPopup to be generic. If you need to pass params to the complete function, you can do it with variables, and not predeterman what variables you pass in.
for example, if the mutation run does not neet sectionId, but rather AgreementId